client/gutenberg/extensions/markdown/editor.js

```diff
@@ -12,47 +12,48 @@ import { registerBlockType } from '@wordpress/blocks';
  * Internal dependencies
  */
 import './editor.scss';
-import edit from './edit';
-import save from './save';
-
-registerBlockType( 'a8c/markdown', {
-	title: __( 'Markdown' ),
-
-	description: (
-		<Fragment>
-			<p>{ __( 'Write your content in plain-text Markdown syntax.' ) }</p>
-			<ExternalLink href="https://en.support.wordpress.com/markdown-quick-reference/">
-				{ __( 'Support reference' ) }
-			</ExternalLink>
-		</Fragment>
-	),
-
-	icon: (
-		<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 208 128">
-			<rect
-				width="198"
-				height="118"
-				x="5"
-				y="5"
-				ry="10"
-				stroke="#000"
-				strokeWidth="10"
-				fill="none"
-			/>
-			<path d="M30 98v-68h20l20 25 20-25h20v68h-20v-39l-20 25-20-25v39zM155 98l-30-33h20v-35h20v35h20z" />
-		</svg>
-	),
-
-	category: 'jetpack',
-
-	keywords: [ __( 'formatting' ), __( 'syntax' ), __( 'markup' ) ],
-
-	attributes: {
-		//The Markdown source is saved in the block content comments delimiter
-		source: { type: 'string' },
-	},
-
-	edit,
-
-	save,
-} );
+
+Promise.all( [ import( './edit' ), import( './save' ) ] ).then(
```

Comment:
It felt a little odd to have the entire block wrapped in a promise. I guess this is for the sake of the experiment, but in real life I'd expect this to be wrapped as some sort of a helper or HoC.